include/openPMD/Series.hpp

```diff
@@ -20,6 +20,10 @@
  */
 #pragma once
 
+// IWYU pragma: private, include <openPMD/openPMD.hpp>
+// IWYU pragma: friend ".*openPMD/.*"
+// IWYU pragma: friend ".*Test\.cpp"
```

Comment:
we could also let out tests include `openPMD/openPMD.hpp` altogether.

Reply:
> Note: Not all headers in include/ are publicly (transitively) included by means of openPMD.hpp.

Could be that I included a few too many now...

Reply:
Without any verification, just the best judgement I can give right now:
```
/* internal only */

#include "openPMD/IO/AbstractFilePosition.hpp"
#include "openPMD/IO/AbstractIOHandlerHelper.hpp"
#include "openPMD/IO/AbstractIOHandler.hpp" // except for the errors
#include "openPMD/IO/AbstractIOHandlerImpl.hpp"

#include "openPMD/IO/ADIOS/ADIOS1Auxiliary.hpp"
#include "openPMD/IO/ADIOS/ADIOS1FilePosition.hpp"
#include "openPMD/IO/ADIOS/ADIOS1IOHandler.hpp"
#include "openPMD/IO/ADIOS/ADIOS1IOHandlerImpl.hpp"
#include "openPMD/IO/ADIOS/ADIOS2FilePosition.hpp"
#include "openPMD/IO/ADIOS/ADIOS2IOHandler.hpp"
#include "openPMD/IO/ADIOS/ParallelADIOS1IOHandler.hpp"
#include "openPMD/IO/ADIOS/ParallelADIOS1IOHandlerImpl.hpp"

#include "openPMD/IO/Format.hpp"

#include "openPMD/IO/HDF5/HDF5Auxiliary.hpp"
#include "openPMD/IO/HDF5/HDF5FilePosition.hpp"
#include "openPMD/IO/HDF5/HDF5IOHandler.hpp"
#include "openPMD/IO/HDF5/HDF5IOHandlerImpl.hpp"
#include "openPMD/IO/HDF5/ParallelHDF5IOHandler.hpp"
#include "openPMD/IO/HDF5/ParallelHDF5IOHandlerImpl.hpp"

#include "openPMD/IO/IOTask.hpp"

#include "openPMD/auxiliary/Filesystem.hpp"
#include "openPMD/auxiliary/Memory.hpp"
#include "openPMD/auxiliary/StringManip.hpp"
```

- [ ] Verify all the claimed private includes

Reply:
That also means we should not *install* those headers :)

Reply:
@C0nsultant could you verify the private-ness of those headers? (how? manually?)

Reply:
Will do so today. And yes, that means going through them manually.

Reply:
@ax3l Done. There are a few headers that are currently publicly exposed without any read need to.